rewrite-maven/src/main/java/org/openrewrite/maven/cache/RocksdbMavenPomCache.java

```diff
@@ -297,6 +300,31 @@ private void fillUnresolvablePoms() {
             } catch (RocksDBException exception) {
                 throw new IllegalStateException("Unable to create cache database." + exception.getMessage(), exception);
             }
+
+            try {
+                cleanCacheIfCorrupt(workspace);
+            } catch (Exception ex) {
+                throw new IllegalStateException("Unable to clear corrupt maven pom cache.", ex);
+            }
+        }
+
+        private void cleanCacheIfCorrupt(String workspace) throws IOException {
+            try {
+                database.verifyChecksum();
+            } catch (RocksDBException ex){
+                if (ex.getMessage().contains("block checksum mismatch")) {
```

Comment:
is it safe to say, that if we get any RocksDBException here that we should just recreate the cache folder? If feels fragile to rely on a specific message.

Reply:
That's a good call out - I'm not completely sure, I was able to get the RocksDBException to be thrown in different ways, but a _corruption_ seemed to be distinguish by the message.

In all fairness, it should be pretty rare and is likely safe to just clear the cache if the exception is thrown at all. I'll update the condition in a moment